core/src/main/java/org/jruby/java/proxies/MapJavaProxy.java

```diff
@@ -438,12 +461,10 @@ public RubyProc to_proc(ThreadContext context) {
         return (RubyProc) newProc;
     }
 
-    /** rb_hash_to_s
-     *
-     */
+    // NOTE: keep Map#to_s -> toString as with other Java types
     @JRubyMethod(name = "to_s")
     public IRubyObject to_s(ThreadContext context) {
-        return getOrCreateRubyHashMap(context.runtime).to_s(context);
+        return RubyString.newString(context.runtime, getMapObject().toString());
```

Comment:
we keep `to_s` -> `toString` for other collections so this seemed inconsistent.
do we want to keep it since we changed `inspect` already to not be a `to_s` alias?

always a bit 2 sided between consistency and "improvements", JRuby re-defs `to_s` on:
- `java.lang.Throwable` - `to_s` -> `message`
- `java.lang.Class` - `to_s` -> `name` (toString is `"class java.lang.String"`)
- on Java arrays as the default `toString` is useless

would be personally inclined towards `java.util.Map` not being wrapped in an internal `Hash` (like the rest of proxies),
in which case `to_s` just ends up as `toString`

Reply:
> would be personally inclined towards java.util.Map not being wrapped in an internal Hash

Me too. I think our goal should be to get MapJavaProxy as close as possible to just patching Map rather than all the places it gives up and turns it into a "special" Hash.

Reply:
If it wasn't clear I support this change. 👍